client/reader/discover/discover-stream.js

```diff
@@ -52,34 +51,16 @@ export const DiscoverHeader = ( props ) => {
 };
 
 const DiscoverStream = ( props ) => {
-	const locale = useLocale();
 	const translate = useTranslate();
 	const followedTags = useSelector( getReaderFollowedTags );
 	const isLoggedIn = useSelector( isUserLoggedIn );
-	const selectedTab = props.selectedTab;
-	const { data: interestTags = [] } = useQuery( {
-		queryKey: [ 'read/interests', locale ],
-		queryFn: () =>
-			wpcom.req.get(
-				{
-					path: `/read/interests`,
-					apiNamespace: 'wpcom/v2',
-				},
-				{
-					_locale: locale,
-				}
-			),
-		select: ( data ) => {
-			return data.interests;
-		},
-	} );
+	const selectedTab = props.selectedTab || DEFAULT_TAB;
+
+	const selectedTag = new URLSearchParams( window.location.search ).get( 'selectedTag' );
 
-	const promptSlug = isBloganuary() ? 'bloganuary' : 'dailyprompt';
-	const promptTitle = isBloganuary() ? translate( 'Bloganuary' ) : translate( 'Daily prompts' );
-	// Add dailyprompt to the front of interestTags if not present.
-	const hasPromptTab = interestTags.filter( ( tag ) => tag.slug === promptSlug ).length;
-	if ( ! hasPromptTab ) {
-		interestTags.unshift( { title: promptTitle, slug: promptSlug } );
```

Comment:
Since we introduced a `DiscoverTagsNavigation` component, I thought it made sense to move this logic there.